### PR TITLE
Add IntelliJ / Android Studio project files.

### PR DESCRIPTION
**Pull request by jasta**
This establishes formatting norms and adds the Copyright enforcement.

### CODE REVIEW

**Comment by jasta** [comment]
@rickbrew can you checkout this branch locally and let me know if IntelliJ explodes?  I'm hoping this is a smooth transition once merged in.

**Comment by brennantaylor** [comment]
Many of the data in the iml files should be inferred from the gradle model. I don't think it is appropriate to commit those files to source.

Also why commit a file that tells intellij that the files are under git version control? It will infer that immediately at startup.

**Comment by jasta** [comment]
@brennantaylor I wondered about this and I think you're quite right.  I was really hoping to just be able to commit the project code style settings and the copyright notice.  Do you know if it's possible to get away with just this bare minimum set and have IntelliJ pick it up?

**Comment by brennantaylor** [comment]
Specifically these files look redundant.

The project `.iml` files also store USER configuration. Like what configuration `buildType` the ui has selected.

````text
.idea/compiler.xml
.idea/encodings.xml
.idea/gradle.xml
.idea/modules.xml
.idea/scopes/scope_settings.xml
.idea/vcs.xml
stetho-okhttp/stetho-okhttp.iml
stetho-parent.iml
stetho-urlconnection/stetho-urlconnection.iml
stetho.iml
stetho/stetho.iml
````

**Comment by brennantaylor** [comment]
@jasta Hmmm. In our project we just have the devs install the codestyle manually. There were bugs in AndroidStudio back in the preview days that forced us to do this. It may be easier now. I can look into it.

**Comment by brennantaylor** [comment]
Based on this doc `.idea/codeStyleSettings.xml` should just work

https://intellij-support.jetbrains.com/entries/23393067

**Comment by jasta** [comment]
Thanks, I'll take a look soon.

**Comment by jasta** [comment]
Removed unnecessary files and tested locally that this works more or less fine.  Android Studio tries to clobber the copyright template on import, but this is apparent and wrong in git and if you simply revert Android Studio's changes it will work properly.

File: .idea/copyright/Facebook.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<component name="CopyrightManager">
+  <copyright>
+    <option name="notice" value="Copyright (c) 2014-present, Facebook, Inc.&#10;All rights reserved.&#10; &#10;This source code is licensed under the BSD-style license found in the&#10;LICENSE file in the root directory of this source tree. An additional grant &#10;of patent rights can be found in the PATENTS file in the same directory." />
```
Comment (longinoa) [inline .idea/copyright/Facebook.xml]: Isnt this supposed to be the current year?

**Comment by longinoa** [comment]
Im not sure about the year part but the rest lgtm

**Pull request merged**